bfcfg

```diff
@@ -197,6 +198,219 @@ mfg_cfg()
+  add_trailing_null ${pad_len} ${tmp_file}
+
+  if [ -n "${MFG_SYS_VER}" ]; then
+    log_msg "mfg: SYS_VER=${MFG_SYS_VER}"
+
+    if [ ${#MFG_SYS_VER} -gt ${sys_ver_max_len} ]; then
+      log_msg "mfg: SYS_VER exceeding max length ${sys_ver_max_len}"
+      return
+    fi
+
+    printf "${MFG_SYS_VER}" >> ${tmp_file} 2>>${log_file}
+    sys_ver_err=$?
+
+    if [ ${sys_ver_err} -eq 0 ]; then
+      log_msg "mfg: sys_ver written"
+    else
+      log_msg "mfg: sys_ver_err=${sys_ver_err}"
+      return
+    fi
+  fi
+  pad_len=$((${sys_ver_max_len}-${#MFG_SYS_VER}))
+  add_trailing_null ${pad_len} ${tmp_file}
+
+  if [ -n "${MFG_BSB_MFR}" ]; then
+    log_msg "mfg: BSB_MFR=${MFG_BSB_MFR}"
+
+    if [ ${#MFG_BSB_MFR} -gt ${bsb_mfr_max_len} ]; then
+      log_msg "mfg: BSB_MFR exceeding max length ${bsb_mfr_max_len}"
+      return
+    fi
+
+    printf "${MFG_BSB_MFR}" >> ${tmp_file} 2>>${log_file}
+    bsb_mfr_err=$?
+
+    if [ ${bsb_mfr_err} -eq 0 ]; then
+      log_msg "mfg: bsb_mfr written"
+    else
+      log_msg "mfg: bsb_mfr_err=${bsb_mfr_err}"
+      return
+    fi
+  fi
+  pad_len=$((${bsb_mfr_max_len}-${#MFG_BSB_MFR}))
+  add_trailing_null ${pad_len} ${tmp_file}
+
+  if [ -n "${MFG_BSB_PDN}" ]; then
+    log_msg "mfg: BSB_PDN=${MFG_BSB_PDN}"
+
+    if [ ${#MFG_BSB_PDN} -gt ${bsb_pdn_max_len} ]; then
+      log_msg "mfg: BSB_PDN exceeding max length ${bsb_pdn_max_len}"
+      return
+    fi
+
+    printf "${MFG_BSB_PDN}" >> ${tmp_file} 2>>${log_file}
+    bsb_pdn_err=$?
+
+    if [ ${bsb_pdn_err} -eq 0 ]; then
+      log_msg "mfg: bsb_pdn written"
+    else
+      log_msg "mfg: bsb_pdn_err=${bsb_pdn_err}"
+      return
+    fi
+  fi
+  pad_len=$((${bsb_pdn_max_len}-${#MFG_BSB_PDN}))
+  add_trailing_null ${pad_len} ${tmp_file}
+
+  if [ -n "${MFG_BSB_VER}" ]; then
+    log_msg "mfg: BSB_VER=${MFG_BSB_VER}"
+
+    if [ ${#MFG_BSB_VER} -gt ${bsb_ver_max_len} ]; then
+      log_msg "mfg: BSB_VER exceeding max length ${bsb_ver_max_len}"
+      return
+    fi
+
+    printf "${MFG_BSB_VER}" >> ${tmp_file} 2>>${log_file}
+    bsb_ver_err=$?
+
+    if [ ${bsb_ver_err} -eq 0 ]; then
+      log_msg "mfg: bsb_ver written"
+    else
+      log_msg "mfg: bsb_ver_err=${bsb_ver_err}"
+      return
+    fi
+  fi
+  pad_len=$((${bsb_ver_max_len}-${#MFG_BSB_VER}))
+  add_trailing_null ${pad_len} ${tmp_file}
+
+  if [ -n "${MFG_BSB_SN}" ]; then
+    log_msg "mfg: BSB_SN=${MFG_BSB_SN}"
+
+    if [ ${#MFG_BSB_SN} -gt ${bsb_sn_max_len} ]; then
+      log_msg "mfg: BSB_SN exceeding max length ${bsb_sn_max_len}"
+      return
+    fi
+
+    printf "${MFG_BSB_SN}" >> ${tmp_file} 2>>${log_file}
+    bsb_sn_err=$?
+
+    if [ ${bsb_sn_err} -eq 0 ]; then
+      log_msg "mfg: bsb_sn written"
+    else
+      log_msg "mfg: bsb_sn_err=${bsb_sn_err}"
+      return
+    fi
+  fi
+  pad_len=$((${bsb_sn_max_len}-${#MFG_BSB_SN}))
+  add_trailing_null ${pad_len} ${tmp_file}
+
+  log_msg "mfgext: bin file create at ${tmp_file}"
+  if [ $no_capsule -eq 0 ]; then
```

Comment:
I think we should set "no_capsule" based on whether bfcfg is run from the platform or from the host. To recall, we need the bfcfg to support input and output parameters (--in and --out). If (--out is specified then set no_capsule to 1).

The --in and --out parameters would allow the provisioning of the MFG/MFG EXT when no OS is present.

```
# bfcfg --in bf.cfg --out bfcfg.bin
# mlx-mkcap --mfg-cfg bfcfg.bin out.cap
# mlx-mkbfb --capsule out.cap default.bfb bfcfg-default.bfb
# cat bfcfg-deafault.bfb > /dev/rshim0/boot
```

If you wish to leave it for later then omit "no_capsule" from the list of options/arguments and assume that capsule is applied by default.

By the way, UEFI shouldn't return any error if the capsule update is initiated on BF2 and BF1. So capsules must be ignored on those platforms until we migrate the MFG.

Reply:
I will leave the --in, --out options for later and will omit "no_capsule" for now.